woocommerce/woocommerce
# Avoid duplicate custom classnames

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR is a follow-up to #50662. The previous PR updated the `StyleAttributeUtils::get_classes_and_styles_by_attributes` method to return custom class names.

However, in some cases, `StyleAttributeUtils` is used alongside `get_block_wrapper_attributes`, which also injects custom class names. This can result in duplicated class names for some blocks. This PR addresses and resolves this issue by ensuring class names are not duplicated.​

Closes [#51883](https://github.com/woocommerce/woocommerce/issues/51883).



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:



1. Visit the Single Product template.
2. Focus the `Add To Cart Form`  block (the next times the indicated block).
3. On the right sidebar, click on advanced classes and add a value.
4. Save.
5. Visit a single product page.
6. Inspect the `Add To Cart Form`.
7. Ensure that the value inserted earlier is present only once in the class names.
8. Repeat 1-7 for the `breacrumbs` block.
9. Remove the `Product Image Gallery` block.
10. Add the `Product Gallery (Beta)`
11. Repeat 1-7 for the `Pager` block (inner block of the `Product Gallery (Beta)` block).
12. Visit the Single Product.
13. Repeat 1-5 for the `Store Notice block`.
14. Save.
15. Visit a single product page.
16. Add the product in the cart.
17.  Ensure that the notice is visible.
18. Inspect the notice.
19. Ensure that the value inserted earlier is present only once in the class names.
20. Visit the Product Catalog template.
21. Repeat 1-7 for the Product Image block.
22. Create a new page.
23. Add the Single Product Block.
24. Select a Product.
25. Add the `Product Average Rating (beta)` block.
26. Repeat 1-7 for the `Product Average Rating (beta)` block.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Avoid duplicate custom classnames

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
